src/components/tl/ucp/reduce_scatter/reduce_scatter_knomial.c

```diff
@@ -17,6 +17,22 @@
         task->reduce_scatter_kn.phase = _phase;                                \
     } while (0)
 
+#define EXEC_TASK_TEST(_phase, _errmsg) do {                                   \
```

Comment:
i would move EXEC_TASK_TEST to tl_ucp_coll.h with "reduce_kn, reduce_scatter_kn, allgather_kn" being another macro argument. Then it is same macro for all colls.